packages/xod-client/src/editor/messages.js

```diff
@@ -21,3 +22,11 @@ export const NO_PATCH_TO_TRANSPILE = composeMessage(
   'No patch opened',
   'Open a patch to upload and try again'
 );
+
+export const deadRefErrorMessage = err => composeMessage(
+  err.title,
+  [
+    getDeadRefErrorReason(err),
```

Comment:
Too much logic for `messages.js`. I’d move it next to `composeMessage` leaving here only very simple text formatting:

```js
deadReference = (deadPatch, pathToDeadPatch) => composeMessage(
  'Patch contains dead references',
  [
    `Patch ${deadPatch} is missing. Required by ${pathToDeadPatch}`,
    `Delete ${deadPatch} nodes  or bring back the patch to continue.`,
  ].join('\n')
);
```